src/ompl/tools/thunder/src/SPARSdb.cpp

```diff
@@ -972,18 +1047,19 @@ bool ompl::geometric::SPARSdb::addStateToRoadmap(const base::PlannerTerminationC
     if (verbose_)
         OMPL_INFORM(" - checkAddCoverage() Are other nodes around it visible?");
     // Coverage criterion
-    if (!checkAddCoverage(qNew,
+    if (denseRoadmap_ || !checkAddCoverage(qNew,
                           visibleNeighborhood))  // Always add a node if no other nodes around it are visible (GUARD)
     {
         if (verbose_)
             OMPL_INFORM(" -- checkAddConnectivity() Does this node connect neighboring nodes that are not connected? ");
         // Connectivity criterion
-        if (!checkAddConnectivity(qNew, visibleNeighborhood))
+        if ((vnbhd.size() > 0) || !checkAddConnectivity(qNew, visibleNeighborhood))
```

Comment:
`> 0` is redundant here

Reply:
You mean that checking (vnbhd.size()) as a boolean has the same effect?